rfc/rfc-0000-k8s-events.md

```diff
@@ -0,0 +1,162 @@
+| StampedObjectInvalid | `<resource name>` could not be applied as `<resource>.<group>/<name>` due to API server error `<error message>` | This object was (probably) malformed. | Owner |
+| StampedObjectInvalid | `<resource name>` could not be applied as `<resource>.<group>` due to API server error `<error message>` | This object was (probably) malformed. This lets operators know their templates might have issues | Blueprint |
+| StampedObjectKindNotFound | `<resource name>` could not be applied because `<resource>.<group>/<name>` does not exist on this cluster | Did someone forget to install the CRDs? Otherwise it's a malformed template | Owner |
+| StampedObjectKindNotFound | `<resource name>` could not be applied because `<resource>.<group>` does not exist on this cluster | Did someone forget to install the CRDs? Otherwise it's a malformed template | Blueprint |
+| StampedObjectApplied | `<resource name>` was applied as `<resource>.<group>/<name>` | a stamped object needed to be created/updated | Owner |
+| StampedObjectKindChanged | `<resource name>` was `<old group>.<old kind>`, now `<new group>.<new kind>` | YTT selection for a template GVK changed, or templated values in GVK changed | Owner |
+| ResourceOutputChanged | `<resource name>` found a new output in `<resource>.<group>/<name>` | a resource produced a new output | Owner |
+| ResourceHealthyStatusChanged | `<resource name>` found a new status in `<resource>.<group>/<name>` | a resource produced a new healthy status | Owner |
+| SupplyChainChanged | supply chain changed from `<old supply chain name>` to `<new supply chain name>` | Workload selected for a new or different supply chain. Note: `none` is a possible name. | Owner |
+| WorkloadSelected | supply chain matched a workload | A workload was selected for (really, this is when the supplyChainRef changes). This makes the count a useful metric | Blueprint |
+
+**Note:** `<resource name>` represents the `blueprint.spec.resource[].name` and *not* `stampedObject.metadata.name`
+
+**Note:** Blueprint event `messages` are usually designed to aggregate on the resource kind, not the individual
+resource (we omit `<resource name>`)
+
+**Note:** On naming, we're tring to start with the location in the object diagram (first diagram) that represents the
+change. `stamped object` is not well known to users, but we need to rectify that, or use a clearer term to describe "the
+resource on etcd", which would normally be "Resource". Unfortunately we've overloaded resource to mean the collective
+concept of a template, a configuration, it's submission to the API and it's resulting changes from external events.
+
+There are a lot of event's here, and more could exist, but we should review these during the RFC review process. I hope
+we can prune the list and make it `as meaningful as is necessary, and no more`
```

Comment:
In Office Hours, there was a discussion that this RFC _not_ define an explicit list of events. Rather that the RFC express a general guideline for events and that the team agree that future RFCs will not be necessary to change the list of event types. Is that a suggestion that is being adopted by this RFC?

Reply:
We've reworded this as a list of examples.